src/ai/backend/manager/services/image/actions/modify_image.py

```diff
@@ -0,0 +1,79 @@
+from dataclasses import dataclass, field
+from typing import Any, Optional, override
+
+from ai.backend.manager.actions.action import BaseActionResult
+from ai.backend.manager.actions.exceptions import BaseActionException
+from ai.backend.manager.data.image.types import ImageData
+from ai.backend.manager.models.image import ImageRow, ImageType
+from ai.backend.manager.services.image.base import ImageAction
+from ai.backend.manager.types import OptionalState, TriState
+
+
+@dataclass
+class ModifyImageInputData:
+    name: OptionalState[str] = field(default_factory=lambda: OptionalState.none("name"))
+    registry: OptionalState[str] = field(default_factory=lambda: OptionalState.none("registry"))
+    image: OptionalState[str] = field(default_factory=lambda: OptionalState.none("image"))
+    tag: OptionalState[str] = field(default_factory=lambda: OptionalState.none("tag"))
+    architecture: OptionalState[str] = field(
+        default_factory=lambda: OptionalState.none("architecture")
+    )
+    is_local: OptionalState[bool] = field(default_factory=lambda: OptionalState.none("is_local"))
+    size_bytes: OptionalState[int] = field(default_factory=lambda: OptionalState.none("size_bytes"))
+    type: OptionalState[ImageType] = field(default_factory=lambda: OptionalState.none("type"))
+    config_digest: OptionalState[str] = field(
+        default_factory=lambda: OptionalState.none("config_digest")
+    )
+    labels: OptionalState[dict[str, Any]] = field(
+        default_factory=lambda: OptionalState.none("labels")
+    )
+    accelerators: TriState[str] = field(default_factory=lambda: TriState.nop("accelerators"))
+    resources: OptionalState[dict[str, Any]] = field(
+        default_factory=lambda: OptionalState.none("resources")
+    )
```

Comment:
It's nice to have field names injected separately from variable names, but it's a bit annoying to have to inject field names at dataclass creation time as well.

Reply:
I agree. Is there any way we can improve it?